src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -175,17 +175,19 @@ export class SamDebugConfigProvider implements vscode.DebugConfigurationProvider
                         getLogger().error(`provideDebugConfigurations: invalid template: ${templateDatum.path}`)
                         continue
                     }
-                    const resources = getResourcesFromTemplateDatum(templateDatum)
-                    for (const resourceKey of resources.keys()) {
-                        const runtimeName = resources.get(resourceKey)?.Properties?.Runtime
-                        configs.push(
-                            createTemplateAwsSamDebugConfig(
-                                folder,
-                                CloudFormation.getStringForProperty(runtimeName, templateDatum.template),
-                                resourceKey,
-                                templateDatum.path
+                    for (const resourceKey of Object.keys(templateDatum.template.Resources)) {
+                        const resource = templateDatum.template.Resources[resourceKey]
```

Comment:
this essentially "inlines" `getResourcesFromTemplateDatum()`, why not keep that function around? Would at least avoid churn.

Reply:
The function shouldn't have existed in the first place...leaving that function around meant we were looping through the data twice (since `getResourcesFromTemplateDatum` represented a full loop through a `templateDatum`).